File: src/visualizer.cpp
#include "ros/ros.h"
#include "geometry_msgs/PointStamped.h"
#include "geometry_msgs/TransformStamped.h"
#include <visualization_msgs/Marker.h>
#include <visualization_msgs/MarkerArray.h>
#include "sawyer_humcpp/mdasys.h"
//#include "std_msgs/Float32MultiArray.h"
//#include <sstream>
//#include <iostream>
//#include <armadillo>
//#include <random>
using namespace std;
class VisualCP{
  ros::Subscriber mda_sub;
  ros::Publisher marker_pub;
  ros::NodeHandle* nh;
  //visualization_msgs::MarkerArray markerset;
  //Mass Marker
  visualization_msgs::Marker mass;
    
   visualization_msgs::Marker marker;std::cout<<"wtf"<<std::endl;
   //marker.header.frame_id = "base_link";
   //marker.header.stamp = ros::Time();
    //mass.header.frame_id = "cart";
    /*mass.header.stamp=ros::Time();
    mass.ns = ros::this_node::getNamespace()+"sim_world";
    mass.id = 0;
    mass.type = visualization_msgs::Marker::SPHERE;
    mass.action=visualization_msgs::Marker::ADD;
    mass.scale.x=0.13 ;mass.scale.y=0.13;mass.scale.z=0.13;
    mass.color.a=1.0;mass.color.r=1.0;mass.color.g=1.0;mass.color.b=1.0;*/
  //Mass Marker
  /*visualization_msgs::Marker link;
    link.header.frame_id = "link";
    link.header.stamp=mass.header.stamp;
    link.ns = ros::this_node::getNamespace()+"sim_world";
    link.id = 1;
    link.type = visualization_msgs::Marker::LINE_STRIP;
    link.action=visualization_msgs::Marker::ADD;
    link.scale.x=0.13 ;mass.scale.y=0.13;mass.scale.z=0.13;
    link.color.a=1.0;mass.color.r=1.0;mass.color.g=1.0;mass.color.b=1.0;
  */
  
  public:
    VisualCP(ros::NodeHandle* _nh){
    nh=_nh;
    ROS_INFO("Starting RVIZ");
    //setup publishers & subscribers
    marker_pub = nh->advertise<visualization_msgs::MarkerArray>("visualization_marker_array", 5);  
    mda_sub = nh->subscribe("/mda_topic",5,&VisualCP::update_state,this);
    };
    void timercall(const ros::TimerEvent& event){
    };
    void update_state(const sawyer_humcpp::mdasys& state){
    };
    
};

int main(int argc, char **argv){
  /*Run the main loop, by instatiating a System class, and then
    calling ros::spin*/
  ros::init(argc, argv, "visualizer");
  ros::NodeHandle n;
  VisualCP marks(&n);
  ros::Timer timer = n.createTimer(ros::Duration(1./60.), &VisualCP::timercall, &marks);
  ros::spin();
 return 0;
};

File: src/visualizer.py
#!/usr/bin/env python
import numpy as np
import copy
import rospy
from std_msgs.msg import ColorRGBA
import visualization_msgs.msg as VM
from geometry_msgs.msg import PointStamped
from sawyer_humcpp.msg import mdasys

SIMFRAME = "trep_world"
MASSFRAME = "pend_mass"
CARTFRAME = "cart"

class Visualizer_CP:
    def __init__(self):
        rospy.loginfo("Initializing Rviz")
        #setup publisher, subscribers,timers:
        self.mda_sub = rospy.Subscriber("mda_topic",mdasys,self.update_marks)
        self.marker_pub = rospy.Publisher("visualization_marker_array", VM.MarkerArray,queue_size=3)
        #self.br = tf.TransformBroadcaster()
        ######SETUP MARKER ARRAY######
        self.markers = VM.MarkerArray()
         # mass marker
        self.mass_marker = VM.Marker()
        self.mass_marker.action = VM.Marker.ADD
        self.mass_marker.color = ColorRGBA(*[1.0, 1.0, 1.0, 1.0])
        self.mass_marker.header.frame_id = rospy.get_namespace() + SIMFRAME 
        self.mass_marker.lifetime = rospy.Duration(5*DT)
        self.mass_marker.scale = GM.Vector3(*[0.13, 0.13, 0.13])
        self.mass_marker.type = VM.Marker.SPHERE
        self.mass_marker.id = 0
        # link marker
        self.link_marker = copy.deepcopy(mass_marker)
        self.link_marker.type = VM.Marker.LINE_STRIP
        self.link_marker.color = ColorRGBA(*[1.0, 0.05, 0.05, 1.0])#0.1, 0.1, 0.,1.0])
        self.link_marker.scale = GM.Vector3(*[0.02, 0.05, 0.05])
        self.link_marker.id = 1
        #cart marker
        self.cart_marker = copy.deepcopy(mass_marker)
        self.cart_marker.type = VM.Marker.CUBE
        self.cart_marker.color = ColorRGBA(*[0.1, 0.5, 1.0, 0.9])
        self.cart_marker.scale = GM.Vector3(*[0.2, 0.2, 0.2])
        self.cart_marker.id = 2
        
        self.markers.markers.append(mass_marker)
        self.markers.markers.append(link_marker)
        self.markers.markers.append(cart_marker)
    def update_marks(self,data):
        rospy.loginfo("got data")
        
def main():
    rospy.init_node('visualizer',log_level=rospy.info
                    
    try:
        disp=Visualizer_CP()
    except rospy.ROsInterruptException: pass
                    
if __name__'__main__':
    main()
                  
        
        
    